source/handbook/all-coding-guidelines.rst

```diff
@@ -0,0 +1,186 @@
+
+``region`` tags should always be surrounded by blank lines with one exception. If they are the first line of a new indent level, a preceding blank line is not required.
+``endregion`` tags should always be surrounded by blank lines.
+
+For example,
+
+.. code-block:: python
+
+    class Something:
+        # region load
+
+        ...
+
+        # endregion
+
+        # region save
+
+        ...
+
+        # endregion
+
+.. note:: In C style languages, the tags are opened and closed with ``//`` single line comments instead of ``#``.
+
+Unit Testing
+************
+
+We prefer to structure unit tests in the Arrange / Act / Assert form in order to clearly delineate the code under test and the assertions being tested.
+Unit tests should contain comments breaking tests into the three sections when they are present.
+This is somewhat of a judgement call as some tests will not have all three sections.
+
+For example,
+
+.. code-block:: python
+
+    # Arrange:
+    wordlist = ['alpha', 'beta', 'gamma', 'delta', 'epsilon', 'eta', 'theta', 'iota', 'kappa', 'lambda']
+    encoder = WordEncoder(wordlist)
+
+    # Act:
+    words = encoder.encode(0)
+
+    # Assert:
+    self.assertEqual(['alpha'], words)
+
+For a more thorough discussion, please review this blog post: https://symbolblog.com/developer-guides/how-to-write-good-unit-tests.
+
+Commenting / Documentation
+**************************
+
+Include a copyright statement at the top of each file. For new projects, use the default copyright contents.
+Copyright and all related rights should be waived for all contributions and via CC0 for all written works.
+
+We document all public methods and classes with at least a sentence or two.
+For projects intended to be used as libraries by external developers, like SDKs, it's recommended to document all parameters.
+For other projects, it's often enough to just reference the parameters without documenting each in detail.
+
+We believe that code should be for the most part self-documenting so use inline comments sparingly.
+Prefer well structured code and good names to inline comments.
+For example, Instead of using comments to delineate the steps of a complicated process, we prefer to refactor the code to have a well-named function per step.
+Inline comments can be used where the purpose of the code is not immediately clear to the reader.
+
+Naming
+******
+
+Avoid unpronounceable names and abbreviations.
+Use a spell checker! ``codespell`` is recommended.
+
+1. User defined types (``class``, ``struct``, ``enum``) should be UpperCamelCase and be nouns. [e.g. ``MyEnum``, ``NodeEndpoint``]
+
+2. Function names should contain verbs [e.g. ``DoSomething``]
+
+3. Boolean properties should generally begin with ``is``, ``has`` or ``should``.
+
+   ``isOpen()`` instead of ``open()``
+
+4. Non-boolean properties should generally avoid ``get`` and ``set`` prefixes:
+
+   ``value(12)`` instead of ``setValue(12)``
+
+   ``value()`` instead of ``getValue()``
+
```

Comment:
not boolean, and not sure if applicable to all languages, but what about `try*` functions/methods?